### ISSUE TITLE
Expand the Source Tree

### ISSUE THREAD

**Issue by jasonLaster** [devtools-html/debugger.html#435]
When we debounced the source tree, the source tree stopped expanding by default

When I logged the tree renders, I noticed that the expanded flag alternated between being true or false. This was surprising as I would have thought that it should be dependent on whether the item has children.

```patch
diff --git a/public/js/components/SourcesTree.js b/public/js/components/SourcesTree.js
index e19e277..d138f9f 100644
--- a/public/js/components/SourcesTree.js
+++ b/public/js/components/SourcesTree.js
@@ -119,6 +119,9 @@ let SourcesTree = React.createClass({
   },
 
   renderItem(item, depth, focused, _, expanded, { setExpanded }) {
+    if (depth == 0) {
+      console.log(item.path, expanded);
+    }
     const arrow = Arrow({
       className: classnames(
         { expanded: expanded,
```

**Comment by jasonLaster** [comment]
Hey, mind taking a look at this @jlongster?

**Comment by jlongster** [comment]
Yep, I should be able to get to it today. Mostly waiting on try for a bunch of bugzilla stuff at this point.

**Comment by jlongster** [comment]
The problem is the fact that we create a new tree every single time because of the "collapsing" process. There is a TODO in there already about this. This is problematic because the identity of the tree items change whenever a new source is added, thus everything loses it's "expanded" flag because it's keyed off of the item instance.

This is not the fault of debouncing the source tree at all, it just exposed it. Now the source tree renders multiple times at start, whereas before it only rendered once after all sources comes in. Multiple renderings & recreating the source tree is causing this. I'll look into a fix.

**Comment by jlongster** [comment]
This is fixed in #510
